File: 프로그래머스/0/120866. 안전지대/안전지대.js
function solution(board) {
    const inRange = function(x, y, n) {
        return 0 <= x && x < n && 0 <= y && y < n;
    }
    
    const n = board.length;
    let bomb = board.map(v => [...v])
    const dxs = [-1, -1, -1, 0, 1, 1, 1, 0];
    const dys = [-1, 0, 1, 1, 1, 0, -1, -1];
    
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (board[i][j] == 1) {
    
                for (let k = 0; k < 8; k++) {
                    x = i + dxs[k], y = j + dys[k];
                    if (inRange(x, y, n)) {
                        bomb[x][y] = 1;
                    }
                }
            }    
        }
    }
        
    return bomb.flat().filter(e => !e).length;
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def in_range(x, y, n):
    return 0 <= x and x < n and 0 <= y and y < n


def solution(board):
    bomb = [[0 for _ in range(len(board))] 
            for _ in range(len(board))]
    dxs, dys = [-1, -1, -1, 0, 1, 1, 1, 0], [-1, 0, 1, 1, 1, 0, -1, -1]
    
    for i in range(len(board)):
        for j in range(len(board)):
            if board[i][j] == 1:
                bomb[i][j] = 1
                
                for dx, dy in zip(dxs, dys):
                    dx, dy = i + dx, j + dy
                    
                    if in_range(dx, dy, len(board)):
                        bomb[dx][dy] = 1
                        
    cnt = 0
    for rows in bomb:
        for elem in rows:
            if elem == 0:
                cnt += 1
                
    return cnt